lib/membrane_mp4/demuxer/isom/samples_info.ex

```diff
@@ -225,7 +247,21 @@ defmodule Membrane.MP4.Demuxer.ISOM.SamplesInfo do
           {delta, [%{sample_count: count - 1, sample_delta: delta} | deltas]}
       end
 
-    {%{size: size, sample_delta: delta},
-     %{track_data | decoding_deltas: deltas, sample_sizes: sample_sizes}}
+    {sample_composition_offset, composition_offsets} =
+      case composition_offsets do
+        [%{sample_count: 1, sample_offset: offset} | composition_offsets] ->
+          {offset, composition_offsets}
+
+        [%{sample_count: count, sample_offset: offset} | composition_offsets] ->
+          {offset, [%{sample_count: count - 1, sample_offset: offset} | composition_offsets]}
+      end
```

Comment:
possibly we could have a private function handling this logic for all cases?

Reply:
For now on I would leave it, as refactor would require further changes.